gensim/models/word2vec.py

```diff
@@ -641,16 +678,16 @@ def scan_vocab(self, sentences, progress_per=10000, trim_rule=None):
             if sentence_no % progress_per == 0:
                 logger.info(
                     "PROGRESS: at sentence #%i, processed %i words, keeping %i word types",
-                    sentence_no, sum(itervalues(vocab)) + total_words, len(vocab)
+                    sentence_no, total_words, len(vocab)
                 )
             for word in sentence:
                 vocab[word] += 1
+                total_words += 1
 
             if self.max_vocab_size and len(vocab) > self.max_vocab_size:
-                total_words += utils.prune_vocab(vocab, min_reduce, trim_rule=trim_rule)
+                utils.prune_vocab(vocab, min_reduce, trim_rule=trim_rule)
```

Comment:
I don't see any tests for this change during pruning, seems risky. Does it really work?

Reply:
hmmm do you really think it needs a new test ? prunce_vocab has not been touched only the counter

Reply:
Yes, definitely. You changed the semantics of how the `total_words` works; for example, the return value of `prune_vocab` is ignored.

It may be correct, but is not obvious to me and deserves a test.